src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java

```diff
@@ -35,45 +33,51 @@ URI getRequestUrl(SteamWebApiRequest request) throws SteamApiException {
 		String scheme = getProtocol();
 		String host = request.getBaseUrl();
 		String path = getRequestPath(request);
-		List<NameValuePair> parameters = getRequestParameters(request.getParameters());
+		String query = getRequestQuery(request.getParameters());
 
-		URI requestUrl = getRequestUri(scheme, host, path, parameters);
-
-		return requestUrl;
+		return getRequestUri(scheme, host, path, query);
 	}
 
-	String getRequestPath(SteamWebApiRequest request) {
-		StringBuilder requestPath = new StringBuilder();
-
-		requestPath.append("/");
-		requestPath.append(request.getApiInterface().toString());
-		requestPath.append("/");
-		requestPath.append(request.getInterfaceMethod().toString());
-		requestPath.append("/");
-		requestPath.append(request.getVersion().toString());
-
-		return requestPath.toString();
+	String getRequestPath(SteamWebApiRequest request) {
+		// creates "/ApiInterface/InterfaceMethod/Version"
+		return Stream.of(
+				request.getApiInterface(),
+				request.getInterfaceMethod(),
+				request.getVersion()
+		)
+		.map(Object::toString)
+		.collect(joining("/", "/", ""));
 	}
 
-	List<NameValuePair> getRequestParameters(Map<String, String> parametersMap) {
-
-		List<NameValuePair> nvps = new ArrayList<>();
+	String getRequestQuery(Map<String, String> parameters) throws SteamApiException {
+		if (getKey() == null)

```

Comment:
Again just a minor nitpick but for uniformity/consistency I would prefer if you used braces even for single-line `if` / `for`